src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +41,193 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
+    s16 phi_s0;
+    s16 phi_s1;
+
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, 0);
+
+    for (i = 0, k = 0; i < 6; i++) {
+        temp_a0 = (i + 1) * 15.f;
+        for (j = 0; j < ARRAY_COUNT(D_80B6E730); j++) {
+            k++;
+            k &= 7;
+
+            spD8.x = D_80B6E730[j] + (s32)(((u32)Rand_Next()) >> 0x1C);
+            spD8.y = ((Rand_ZeroOne() - 0.5f) * 15.0f) + temp_a0;
+            spD8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+
+            spCC.x = (2.0f * (Rand_ZeroOne() - 0.5f)) + (spD8.x * 0.018461538f);
+            spCC.y = (Rand_ZeroOne() * 7.0f) + 4.0f;
+            spCC.z = spD8.z * 0.3f;
+
+            Matrix_MultVec3f(&spD8, &pos);
+            Matrix_MultVec3f(&spCC, &velocity);
+
+            pos.x += this->dyna.actor.world.pos.x;
+            pos.y += this->dyna.actor.world.pos.y;
+            pos.z += this->dyna.actor.world.pos.z;
+
+            if (Rand_Next() % 4 == 0) {
+                phi_s0 = 0x20;
+            } else {
+                phi_s0 = 0x40;
+            }
+
+            if (k < 2 || Rand_Next() > 0) {
+                phi_s0 |= 1;
+                phi_s1 = 1;
+                func_800B0E48(globalCtx, &pos, &gZeroVec3f, &dustAccel, &primColor, &envColor,
+                              (((u32)Rand_Next() >> 0x1B) + 70),
+                              ((((u32)Rand_Next()) >> 0x1A) + 60)); // for dust spawn
+            } else {
+                phi_s1 = 0;
+            }
+            EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -550, phi_s0, 30, 0, 0, scales[k], phi_s1, 0, 50, -1,
+                                 OBJECT_KIN2_OBJ, D_06000128);
+        }
+    }
+}
+
+void BgKin2Bombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = THIS;
+    ColliderCylinder* bombwallCollider;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    bombwallCollider = &this->collider;
+    Collider_InitCylinder(globalCtx, bombwallCollider);
+    if (Flags_GetSwitch(globalCtx, BG_KIN2_BOMBWALL_SWITCH_FLAG(this))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &D_06000490);
+    Collider_SetCylinder(globalCtx, bombwallCollider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, bombwallCollider);
+    Actor_SetFocus(&this->dyna.actor, 60.0f);
+    BgKin2Bombwall_SetupWait(this);
+}
+
+void BgKin2Bombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgKin2Bombwall_SetupWait(BgKin2Bombwall* this) {
+    this->actionFunc = BgKin2Bombwall_Wait;
+}
+
+void BgKin2Bombwall_Wait(BgKin2Bombwall* this, GlobalContext* globalCtx) {
+    if (BgKin2Bombwall_IsHitFromNearby(this, globalCtx)) { // checks if AC collision happened
```

Comment:
Comment unnecessary, it's easy to scroll up and look at the function itself (and if you did want to document what it does, put it in a multiline comment above the actual function).

Reply:
Comment removed. Change will be visible in next pushed commit.